reconcile/rhidp/ocm_oidc_idp/base.py

```diff
@@ -82,138 +88,134 @@ def run(
+        )
+        try:
+            oauth_data = secret_reader.read_all_secret(secret)
+        except Exception:
+            logging.warning(
+                f"Unable to read secret in path {secret.path}. "
+                f"Maybe not created yet? Skipping OIDC config for cluster {cluster.name}"
             )
-            gc = (
-                auth.claims.groups
-                if auth.claims and auth.claims.groups
-                else DEFAULT_GROUPS_CLAIMS
+            continue
+
+        sso_client = SSOClient(**oauth_data)
+        if sso_client.issuer != cluster.auth.issuer:
+            # this can only happen if someone manually change the secret or copied it
+            logging.error(
+                f"SSO client issuer {sso_client.issuer} does not match configured cluster issuer "
+                f"{cluster.auth.issuer}. Skipping OIDC config for cluster {cluster.name}"
             )
-            desired_state.append(
-                OCMOidcIdp(
-                    cluster=cluster.name,
-                    name=auth.name,
-                    client_id=sso_client.client_id,
-                    client_secret=sso_client.client_secret,
-                    issuer=auth.issuer,
-                    email_claims=ec,
-                    name_claims=nc,
-                    username_claims=uc,
-                    groups_claims=gc,
-                )
+            continue
+        desired_state.append(
+            IDPState(
+                cluster=cluster,
+                idp=OCMOIdentityProviderOidc(
+                    name=cluster.auth.name,
+                    open_id=OCMOIdentityProviderOidcOpenId(
+                        client_id=sso_client.client_id,
+                        client_secret=sso_client.client_secret,
+                        issuer=cluster.auth.issuer,
+                    ),
+                ),
             )
+        )
 
     return desired_state
 
 
 def act(
     dry_run: bool,
-    ocm_map: OCMMap,
-    current_state: Sequence[OCMOidcIdp],
-    desired_state: Sequence[OCMOidcIdp],
+    ocm_api: OCMBaseClient,
+    current_state: Sequence[IDPState],
+    desired_state: Sequence[IDPState],
     managed_idps: list[str],
 ) -> None:
     """Compare current and desired OIDC identity providers and add, remove, or update them."""
-    to_add = set(desired_state) - set(current_state)
-    to_remove = set(current_state) - set(desired_state)
-    to_compare = set(current_state) & set(desired_state)
-
-    for idp in to_remove:
-        if managed_idps and idp.name not in managed_idps:
-            logging.debug(f"Skipping removal of unmanged '{idp.name}' IDP.")
-            continue
-        logging.info(["remove_oidc_idp", idp.cluster, idp.name])
-        if not idp.id:
-            logging.error(
-                "No identity provider id was given. This should never ever happen!"
+    diff_result = diff_iterables(
+        current_state,
+        desired_state,
+        key=lambda idp_state: hash(
```

Comment:
tuple should be enough here, like

```python
        key=lambda idp_state: (
            idp_state.cluster.organization_id,
            idp_state.cluster.name,
            idp_state.idp.type,
            idp_state.idp.name,
        ),

```

Reply:
Applied in [f1a1942](https://github.com/app-sre/qontract-reconcile/pull/3775/commits/f1a1942cbf1e3a0a3b438ea3613e9e1187b62d37)